pkg/commonvpc/vpc.go

```diff
@@ -153,14 +153,25 @@ func (v *VPC) Subnets() []*Subnet {
 	return v.SubnetsList
 }
 
+// SubnetExpose - can the subnet be exposed to the public internet.
+// out of the platforms we support at the moment, this is used in AWS and not in IBM
+// In AWS each subnet is private or public and only the latter can connect to/from the public internet
+// for IBM the value is set to the default - dontCare (in IBM all subnets can connect to the public internet)
+type SubnetExpose int
+
+const (
+	dontCareExpose SubnetExpose = iota
+	privateExpose
+	publicExpose
+)
+
 // Subnet implements vpcmodel.Subnet interface
 type Subnet struct {
 	vpcmodel.VPCResource
-	VPCnodes []vpcmodel.Node  `json:"-"`
-	Cidr     string           `json:"-"`
-	IPblock  *ipblock.IPBlock `json:"-"`
-	// isPrivate is relevant only for aws, the user set for each subnet if it public (i.e. - has access to the internet)
-	isPrivate bool `json:"-"`
+	VPCnodes     []vpcmodel.Node  `json:"-"`
+	Cidr         string           `json:"-"`
+	IPblock      *ipblock.IPBlock `json:"-"`
+	subnetExpose SubnetExpose     `json:"-"`
```

Comment:
```suggestion
	subnetExpose SubnetExpose
```
No need to prevent json marshaling if a data member is not exported (starts with a small letter)

Reply:
fixed